astropy/time/tests/test_delta.py

```diff
@@ -100,6 +100,27 @@ def test_add(self):
         t2 = dt + self.t
         assert t2.iso == self.t2.iso
 
+    def test_time_delta_comp_num_quantity(self):
+        dt1 = TimeDelta(1, format="sec")
+        dt2 = 2 * u.s
+        assert dt1 < dt2
+        assert dt1 <= dt2
+        assert not dt1 > dt2
+        assert not dt1 >= dt2
+        assert dt1 != dt2
+        assert not dt1 == dt2
+
+    def test_time_delta_comp_nan_quantity(self):
+        # see https://github.com/astropy/astropy/issues/15230
+        dt1 = TimeDelta(1, format="sec")
+        dt2 = np.nan * u.s
+        assert not dt1 < dt2
+        assert not dt1 <= dt2
+        assert not dt1 > dt2
```

Comment:
What happens if the order is reversed? Everything still working?

Reply:
You mean inverting left and right operands ? I didn't think about it, I'll update the tests to cover all cases

Reply:
done